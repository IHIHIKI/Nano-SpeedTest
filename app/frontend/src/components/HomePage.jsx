import React, { Component } from 'react';
import Header from './Header';
import Ad from './Ad';
import AdvancedModal from './AdvancedModal';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { switchTab } from '../actions/navigation';
import { fetchRandomTransaction } from '../actions/table';
import '../styles/HomePage.css';

class HomePage extends Component {
    // component specific state -- don't need to add to redux store
    state = {
        modalOpen: false
    };

    onAdvancedClick = () => {
      this.setState(() => ({modalOpen: true}));
    };

    onClick = () => {
        // navigate to /Stats route
        this.props.history.push('/Stats');
        this.props.onGoPressed(); // Update current active tab and dispatch action to get transaction data
    };

    handleLocationSettings = (e) => {
        // don't reload page on form submit from modal
        e.preventDefault();
        this.setState(() => ({modalOpen: false}));
    };

    handleMultiSettings = (e) => {
        // don't reload page on form submit from modal
        e.preventDefault();
        this.setState(() => ({modalOpen: false}));

        console.log(this.props.advSettings.values);
    };

    handleCancel = () => {
        this.setState(() => ({modalOpen: false}));
    };

    render() {
        return (
            <div className='HomePage'>
                <Header/>
                <Ad/>
                <AdvancedModal
                    open={this.state.modalOpen}
                    handleLocationSettings={this.handleLocationSettings}
                    handleMultiSettings={this.handleMultiSettings}
                    handleCancel={this.handleCancel}
                />
                <h1 className='greeting page-header text-center'>Welcome to NanoSpeed.live!</h1>
                <div className='container'>
                    <div className='row'>
                        <div className='col-md-12 text-center'>
                            <button type='button' className='btn btn-success btn-circle btn-xl' onClick={this.onClick}>
                                Go
                            </button>
                            <br/>
                            <button id='advanced-btn' type='button' className='btn btn-primary' onClick={this.onAdvancedClick}>Advanced</button>
                            <br/>
                            <p className='greeting'>Run your live test now.</p>
                        </div>
                    </div>
                    <div/>
                </div>
            </div>
        );
    }
}
const mapStateToProps = (state) => {
    return {
        advSettings: state.form.advSettings,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        onGoPressed() {
            dispatch(fetchRandomTransaction());
            dispatch(switchTab('Stats')); // Update current active tab
        }
    };
};

HomePage.propTypes = {
    history: PropTypes.object.isRequired,
    onGoPressed: PropTypes.func.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(HomePage);