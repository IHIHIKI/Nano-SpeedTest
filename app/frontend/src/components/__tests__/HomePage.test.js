import React from 'react';
import renderer from 'react-test-renderer';
import HomePage from '../HomePage';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';

const mockStore = configureStore();
let store = mockStore({
    navigation: {
        activeTab: ''
    }
});

const props = {
    history: {},
    onGoPressed: () => {},
    advSettingsForm: {},
    nodes: {
        1: {
            location: 'A'
        },
        2: {
            location: 'B'
        }
    }
};

it('shallow renders without crashing', () => {
    shallow(<HomePage store={store} {...props}/>);
});

it(`renders base snapshot correctly`, () => {
    const tree = renderer.create(
        <Provider store={store}>
            <BrowserRouter>
                <HomePage {...props}/>
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it(`renders multi city snapshot correctly`, () => {
    props.advSettings = {
        values: {
            numTransactions: 3
        }
    };

    const tree = renderer.create(
        <Provider store={store}>
            <BrowserRouter>
                <HomePage {...props}/>
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it(`renders two city snapshot correctly`, () => {
    props.advSettings = {
        values: {
            origin: 1,
            destination: 2
        }
    };

    const tree = renderer.create(
        <Provider store={store}>
            <BrowserRouter>
                <HomePage {...props}/>
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
