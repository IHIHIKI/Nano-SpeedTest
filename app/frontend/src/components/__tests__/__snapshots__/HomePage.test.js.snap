// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders base snapshot correctly 1`] = `
<div
  className="HomePage"
>
  <div
    className="nav nav-tabs"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
         
        Speed Test
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Results"
        onClick={[Function]}
      >
         
        Your Results
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Stats"
        onClick={[Function]}
      >
         
        Global Statistics
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Info"
        onClick={[Function]}
      >
         
        More Info
         
      </a>
    </li>
  </div>
  <div
    className="container"
  >
    <div
      className="jumbotron"
    >
      <div
        id="ad-content"
      >
        <a
          className="row"
          href="https://nano21.net"
        >
          <b>
            ♠ Nano21.net 
          </b>
            | 20 Nano Give-Away for testers! | Nano Blackjack Casino
        </a>
        <p
          className="row"
        >
          COMMUNITY AD
        </p>
      </div>
    </div>
  </div>
  <h1
    className="greeting page-header text-center"
  >
    Welcome to NanoSpeed.live!
  </h1>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-12 text-center"
      >
        <h3
          className="greeting"
        >
          Hit GO to send Nano between two random nodes!
        </h3>
        <button
          className="btn btn-success btn-circle btn-xl"
          onClick={[Function]}
          type="button"
        >
          Go
        </button>
        <br />
        <button
          className="btn btn-primary"
          id="advanced-btn"
          onClick={[Function]}
          type="button"
        >
          Advanced
        </button>
        <br />
        <p
          className="greeting"
        >
          Run your live test now.
        </p>
      </div>
    </div>
    <div />
    <div
      className="shadow-none p-3 mb-5 bg-light rounded"
    >
      <h5>
        What is Nano?
      </h5>
      Nano is a next-generation cryptocurrency created by Colin LeMahieu for instant and free transactions. It's block-lattice structure enables decentralized transactions without loss of security, speed, or high costs.
    </div>
  </div>
</div>
`;

exports[`renders multi city snapshot correctly 1`] = `
<div
  className="HomePage"
>
  <div
    className="nav nav-tabs"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
         
        Speed Test
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Results"
        onClick={[Function]}
      >
         
        Your Results
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Stats"
        onClick={[Function]}
      >
         
        Global Statistics
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Info"
        onClick={[Function]}
      >
         
        More Info
         
      </a>
    </li>
  </div>
  <div
    className="container"
  >
    <div
      className="jumbotron"
    >
      <div
        id="ad-content"
      >
        <a
          className="row"
          href="https://nano21.net"
        >
          <b>
            ♠ Nano21.net 
          </b>
            | 20 Nano Give-Away for testers! | Nano Blackjack Casino
        </a>
        <p
          className="row"
        >
          COMMUNITY AD
        </p>
      </div>
    </div>
  </div>
  <h1
    className="greeting page-header text-center"
  >
    Welcome to NanoSpeed.live!
  </h1>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-12 text-center"
      >
        <h3
          className="greeting"
        >
          Hit GO to send Nano between two random nodes!
        </h3>
        <button
          className="btn btn-success btn-circle btn-xl"
          onClick={[Function]}
          type="button"
        >
          Go
        </button>
        <br />
        <button
          className="btn btn-primary"
          id="advanced-btn"
          onClick={[Function]}
          type="button"
        >
          Advanced
        </button>
        <br />
        <p
          className="greeting"
        >
          Run your live test now.
        </p>
      </div>
    </div>
    <div />
    <div
      className="shadow-none p-3 mb-5 bg-light rounded"
    >
      <h5>
        What is Nano?
      </h5>
      Nano is a next-generation cryptocurrency created by Colin LeMahieu for instant and free transactions. It's block-lattice structure enables decentralized transactions without loss of security, speed, or high costs.
    </div>
  </div>
</div>
`;

exports[`renders two city snapshot correctly 1`] = `
<div
  className="HomePage"
>
  <div
    className="nav nav-tabs"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
         
        Speed Test
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Results"
        onClick={[Function]}
      >
         
        Your Results
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Stats"
        onClick={[Function]}
      >
         
        Global Statistics
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Info"
        onClick={[Function]}
      >
         
        More Info
         
      </a>
    </li>
  </div>
  <div
    className="container"
  >
    <div
      className="jumbotron"
    >
      <div
        id="ad-content"
      >
        <a
          className="row"
          href="https://nano21.net"
        >
          <b>
            ♠ Nano21.net 
          </b>
            | 20 Nano Give-Away for testers! | Nano Blackjack Casino
        </a>
        <p
          className="row"
        >
          COMMUNITY AD
        </p>
      </div>
    </div>
  </div>
  <h1
    className="greeting page-header text-center"
  >
    Welcome to NanoSpeed.live!
  </h1>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-12 text-center"
      >
        <h3
          className="greeting"
        >
          Hit GO to send Nano between two random nodes!
        </h3>
        <button
          className="btn btn-success btn-circle btn-xl"
          onClick={[Function]}
          type="button"
        >
          Go
        </button>
        <br />
        <button
          className="btn btn-primary"
          id="advanced-btn"
          onClick={[Function]}
          type="button"
        >
          Advanced
        </button>
        <br />
        <p
          className="greeting"
        >
          Run your live test now.
        </p>
      </div>
    </div>
    <div />
    <div
      className="shadow-none p-3 mb-5 bg-light rounded"
    >
      <h5>
        What is Nano?
      </h5>
      Nano is a next-generation cryptocurrency created by Colin LeMahieu for instant and free transactions. It's block-lattice structure enables decentralized transactions without loss of security, speed, or high costs.
    </div>
  </div>
</div>
`;
